api/src/api/diag.ts

```diff
@@ -117,7 +93,47 @@ export class DiagAPI implements DiagLogger {
    * @param logLevel - [Optional] The DiagLogLevel used to filter logs sent to the logger. If not provided it will default to INFO.
    * @returns true if the logger was successfully registered, else false
    */
-  public setLogger!: (logger: DiagLogger, logLevel?: DiagLogLevel) => boolean;
+  public setLogger(logger: DiagLogger, logLevel?: DiagLogLevel): boolean;
```

Comment:
NOOOOO!

Do not define as a public function, this causes the generated code to become 
DiagAPI.prototype.setLogger() (which is mostly NOT minifiable), 

it also requires that (if) you have any private functions that MUST be also defined on the class and referenced as this.xxxx, as opposed to being hidden in the closure and therefore more compressible

Reply:
Should be able to still make it an overload I think. The overload is just typing.

Reply:
```typescript
// types.ts
export interface LoggerOptions {
  logLevel?: DiagLogLevel;
  suppressOverrideMessage?: boolean;
}

export interface DiagLoggerApi {
  /**
   * Set the global DiagLogger and DiagLogLevel.
   * If a global diag logger is already set, this will override it.
   *
   * @param logger - [Optional] The DiagLogger instance to set as the default logger.
   * @param options - [Optional] And object which can contain `logLevel: DiagLogLevel` used to filter logs sent to the logger. If not provided it will default to INFO. The object may also contain `suppressOverrideMessage: boolean` which will suppress the warning normally logged when a logger is already registered.
   * @returns true if the logger was successfully registered, else false
   */
   setLogger(logger: DiagLogger, options?: LoggerOptions): boolean;

   /**
    * Set the global DiagLogger and DiagLogLevel.
    * If a global diag logger is already set, this will override it.
    *
    * @param logger - [Optional] The DiagLogger instance to set as the default logger.
    * @param options - [Optional] And object which can contain `logLevel: DiagLogLevel` used to filter logs sent to the logger. If not provided it will default to INFO. The object may also contain `suppressOverrideMessage: boolean` which will suppress the warning normally logged when a logger is already registered.
    * @returns true if the logger was successfully registered, else false
    */
   setLogger(logger: DiagLogger, logLevel?: DiagLogLevel): boolean;
 }
```

```typescript
// diag.ts

export class DiagAPI implements DiagLogger, DiagLoggerApi {
  // some methods and properties elided

  public setLogger!: DiagLoggerApi["setLogger"];
}
```

Reply:
I'll give it a try tomorrow (CET). Feel free to push to this branch in the meantime 🙂

Reply:
Updated, thanks! 👍